ballerina/utils.bal

```diff
@@ -15,21 +15,154 @@
-} external;
+isolated function getUUIDFromBytes(byte[] uuid) returns string {
+    int msb = ((uuid[0] & 0xFF) << 56) |
+            ((uuid[1] & 0xFF) << 48) |
+            ((uuid[2] & 0xFF) << 40) |
+            ((uuid[3] & 0xFF) << 32) |
+            ((uuid[4] & 0xFF) << 24) |
+            ((uuid[5] & 0xFF) << 16) |
+            ((uuid[6] & 0xFF) << 8) |
+            ((uuid[7] & 0xFF) << 0);
+
+    int lsb = ((uuid[8] & 0xFF) << 56) |
+            ((uuid[9] & 0xFF) << 48) |
+            ((uuid[10] & 0xFF) << 40) |
+            ((uuid[11] & 0xFF) << 32) |
+            ((uuid[12] & 0xFF) << 24) |
+            ((uuid[13] & 0xFF) << 16) |
+            ((uuid[14] & 0xFF) << 8) |
+            ((uuid[15] & 0xFF) << 0);
+
+    return getUuidFromSignificantBits(msb, lsb);
+}
+
+isolated function getMostSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
+
+    int mostSigBits = u.timeLow & 0xffffffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeMid & 0xffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeHiAndVersion & 0xffff;
+
+    return mostSigBits;
+}
+
+isolated function getLeastSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
+
+    int leastSigBits;
+    string clockSeq = regex:split(uuid, "-")[3];
+    int|error clockSeqInt = ints:fromHexString(clockSeq);
+    if clockSeqInt is int {
+        leastSigBits = clockSeqInt & 0xffff;
+    } else {
+        return error Error("Failed to get clock sequence value of the uuid");
+    }
+
+    leastSigBits <<= 48;
+    leastSigBits |= u.node & 0xffffffffffff;
+
+    return leastSigBits;
+}
+
+isolated function get64LeastSignificantBitsForVersion1() returns int {
+    int random63BitLong = nextLong() & 0x3FFFFFFFFFFFFFFF;
+    int variant3BitFlag = -0x8000000000000000;
+    return random63BitLong + variant3BitFlag;
+}
+
+isolated function get64MostSignificantBitsForVersion1() returns int {
+    // The gregorian time(1582-10-15 0:0:0) is a constant
+    int gregorianTimeInSeconds = 12219292800;
+    time:Utc duration = time:utcAddSeconds(time:utcNow(), <decimal>gregorianTimeInSeconds);
+    int timeForUuidIn100Nanos = duration[0] * 10000000 + <int>duration[1] * 100;
+    int least12SignificantBitOfTime = (timeForUuidIn100Nanos & 0x000000000000FFFF) >> 4;
+    int uuidVersion = 1 << 12;
+    return (timeForUuidIn100Nanos & 0x7FFFFFFFFFFF0000) + uuidVersion + least12SignificantBitOfTime;
+}
+
+isolated function formatUnsignedInt(int val, int shift, byte[] buf, int offset, int len) returns byte[] {
+    int result = val;
+    int charPos = offset + len - 1;
+    int radix = 1 << shift;
+    int mask = radix - 1;
+
+    while (charPos > offset - 1) {
+        lock {
+            buf[charPos] = digits[result & mask].toBytes()[0];
+        }
+        result >>>= shift;
+        charPos -= 1;
+    }
+    return buf;
+}
+
+isolated function getBytesFromSignificantBits(int msb, int lsb) returns byte[] {
```

Comment:
Can we just say `bitsToBytes` here?

Reply:
Fixed in [03747ed](https://github.com/ballerina-platform/module-ballerina-uuid/pull/376/commits/03747ed1608a8a8c02e488a630f335980ea719cc)